Workshop.md

```diff
@@ -0,0 +1,479 @@
+
+By default, web servers only serve content to web pages that were served by that web server. In order to allow other web pages, such as the ToDo-Backend test page, to connect to the server, [Cross Origin Resource Sharing (CORS)](https://en.wikipedia.org/wiki/Cross-origin_resource_sharing) must be enabled.
+
+Kitura provides a package which makes it easy to enable CORS in your application. Let's add CORS to your project.
+
+1. Open the `ToDoServer` > `Package.swift` file
+2. Add the following to the end of the dependencies section of the Package.swift file:
+    ```swift
+        .package(url: "https://github.com/IBM-Swift/Kitura-CORS.git", from: "2.1.0"),
+    ```
+3. Update the target dependencies for the "Application" target to the following (note the lack of hyphen in KituraCORS):
+    ```swift
+    .target(name: "Application", dependencies: ["KituraCORS", "KituraOpenAPI", "Kitura", "Configuration", "CloudEnvironment", "Health", "SwiftMetrics" ]),
+    ```
+
+In order for Xcode to pick up the new dependency, the Xcode project now needs to be regenerated.
+
+1. Close Xcode.
+2. Regenerate the Xcode project and reopen:
+    ```
+    cd ~/ToDoBackend/ToDoServer  
+    swift package generate-xcodeproj
+    open ToDoServer.xcodeproj
+    ```
+
+Now we need to enable CORS in our Kitura server.
+
+1. Open the `Sources` > `Application` > `Application.swift` file
+2. Add an import for the CORS library to the start of the file:
+   ```swift
+   import KituraCORS
+   ```
+3. Add the following code at the start of the `postInit()` function:  
+   ```swift
+   let options = Options(allowedOrigin: .all)
+   let cors = CORS(options: options)
+   router.all("/*", middleware: cors)
+   ```
+
+4. Re-run the server project in Xcode  
+    1) Edit the scheme again and select a Run Executable of "ToDoServer". 2) Run the project, then "Allow incoming network connections" if you are prompted.
+
+5. Re-run the tests by reloading the test page in your web browser. 
+
+The first test should now be passing!  But the second test is failing:
+
+:x: `the api root responds to a POST with the todo which was posted to it`
+
+In order to fix this, we need to implement a `POST` request that saves a ToDo item.
+
+### 4. Add Support for handling a POST request on `/`
+
+REST APIs typically consist of an HTTP request using a verb such as `POST`, `PUT`, `GET` or `DELETE` along with a URL and an optional data payload. The server then handles the request and responds with an optional data payload.
+
+A request to store data typically consists of a POST request with the data to be stored, which the server then handles and responds with a copy of the data that has just been stored. This means we need to define a `ToDo` type, register a  handler for POST requests on `/`, and implement the handler to store the data.
+
+1. Define a data type for the ToDo items:
```

Comment:
The sub-numbering in line 195 (and indeed in a few other places in this file) looks ok in the source but renders with a combination of letters and numbers. :-(